src/data/mod.rs

```diff
@@ -0,0 +1,46 @@
+/*
+ * This file is part of Astarte.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::io::Error;
+
+use async_trait::async_trait;
+use tokio::sync::mpsc::Receiver;
+use tonic::Status;
+
+use crate::astarte_message_hub::AstarteNode;
+use crate::proto_message_hub::AstarteMessage;
+
+mod astarte;
+#[cfg(test)]
+mod mock_astarte;
+
+#[async_trait]
+pub trait AstartePublisher: Send + Sync {
```

Comment:
I'm not sure about this: shouldn't we have a .rs file for each trait?

Reply:
We use separated .rs file for each of `impl trait`.